gradle/compile.gradle

```diff
@@ -31,7 +31,9 @@ if (java7Home) {
     javaGroovyCompileTasks.all {
         options.with {
             fork = true
-            forkOptions.javaHome = jvmForCompilation.javaHome
+            if (!(project in rootProject.findProperty('projectsRequiresJava8'))) {
+                forkOptions.javaHome = jvmForCompilation.javaHome
```

Comment:
If I remember correctly, on CI we run the Gradle build always with a Java 8 JVM, since we are using kotlin DSL.
We still want to compile most of Gradle with Java 7 to make sure it still works on Java 7.

Not setting `forkOptions.javaHome` to a Java 7 JDK means that the project will be built with the Java 8 JDK Gradle is running with. As long as everyone uses a Java 8 JDK to run Gradle caching will still work.

So I think we don't have a problem here.

Reply:
Cool, thanks!